internal/pkg/server/agent.go

```diff
@@ -385,5 +381,11 @@ func (a *Agent) configFromUnits() (*config.Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	return config.FromConfig(cfgData)
+
+	cliCfg := ucfg.MustNewFrom(a.cliCfg, config.DefaultOptions...)
+	err = cliCfg.Merge(cfgData, config.DefaultOptions...)
```

Comment:
I think we can probably unit test this to avoid future regressions here.

Was this problem introduced in v2 or was it always here?

Reply:
i think it's introduced but i cannot tell for sure.